File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/rest/IngredientModifyRequest.java
package com.tarbadev.witchcraft.recipes.rest;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class IngredientModifyRequest {
  private Integer id;
  private String name;
  private Double quantity;
  private String unit;
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/recipes/rest/entity/IngredientModifyRequest.kt
package com.tarbadev.witchcraft.recipes.rest.entity

import com.tarbadev.witchcraft.converter.UnitHelper.getQuantity
import com.tarbadev.witchcraft.recipes.domain.entity.Ingredient

data class IngredientModifyRequest(
    val id: Int = 0,
    val name: String = "",
    val quantity: Double = 0.0,
    val unit: String = ""
) {
  fun toIngredient(): Ingredient {
    return Ingredient(
        id = id,
        name = name,
        quantity = getQuantity(quantity, unit)
    )
  }
}